From e4e68739737b19c41efbb5ab508cfdd163c2536f Mon Sep 17 00:00:00 2001
From: CismonX <admin@cismon.net>
Date: Tue, 7 May 2024 11:22:59 +0000
Subject: [PATCH] fuse_common.h: fix warning on _Static_assert() (#939)

_Static_assert() is an ISO C11 feature.  Make the check more
standard-conformant so that the compiler won't give pedantic warnings.
---
 include/fuse_common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/fuse_common.h b/include/fuse_common.h
index 4efa53d..bafcda0 100644
--- a/include/fuse_common.h
+++ b/include/fuse_common.h
@@ -957,7 +957,7 @@ void fuse_loop_cfg_convert(struct fuse_loop_config *config,
  * On 32bit systems please add -D_FILE_OFFSET_BITS=64 to your compile flags!
  */
 
-#if defined(__GNUC__) && (__GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ >= 6) && !defined __cplusplus
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)
 _Static_assert(sizeof(off_t) == 8, "fuse: off_t must be 64bit");
 #else
 struct _fuse_off_t_must_be_64bit_dummy_struct \
-- 
2.46.0.rc0.windows.1

